### ISSUE TITLE
BCFtools sort zero position overflow

### ISSUE THREAD

**Issue by Rapsssito** [samtools/bcftools#1753]
There is an overflow issue when sorting a VCF file with zero positions. Here is an example:

### Unsorted VCF
```
MT	0	1915430843:1	N	]MT:16640]N	99	PASS	.
MT	0	3110098432:1	N	]MT:16638]N	99	PASS	.
MT	16638	3110098432:2	G	G[MT:0[	99	PASS	.
MT	16640	1915430843:2	G	G[MT:0[	99	PASS	.
```

#### Commands
```
bcftools sort test.vcf -o test.sorted.vcf
```

### Sorted VCF
```
MT	4294967296	3110098432:1	N	]MT:16638]N	99	PASS	.
MT	4294967296	1915430843:1	N	]MT:16640]N	99	PASS	.
MT	16638	3110098432:2	G	G[MT:0[	99	PASS	.
MT	16640	1915430843:2	G	G[MT:0[	99	PASS	.
```

Note how the `0` from `3110098432:1` has been wrongly changed to `4294967296`.

**Comment by Rapsssito** [comment]
@pd3, could you check if you can reproduce this issue?

**Comment by Rapsssito** [comment]
@pd3, I have created the corresponding issue in htslib: https://github.com/samtools/htslib/issues/1475

**Comment by daviesrob** [comment]
Fixed by samtools/htslib#1476
